Valid Parentheses/step3-using-stack.py

```diff
@@ -0,0 +1,16 @@
+class Solution:
+    def isValid(self, s: str) -> bool:
+        stack = []
+        bracket_pairs = { "}": "{", "]": "[", ")": "("}
+
+        for char in s:
+            if char in bracket_pairs:
+                if len(stack) > 0 and stack[-1] == bracket_pairs[char]:
+                    stack.pop()
+                else:
+                    return False
+            else:
+                stack.append(char)
+        
+        return stack == []
```

Comment:
こちらは空のリストはfalseであることを利用して以下のようにも書けますね
```suggestion
        return not stack
```

ご参考までにGoogleのスタイルガイドではこの書き方が推奨されているようです
> For sequences (strings, lists, tuples), use the fact that empty sequences are false, so if seq: and if not seq: are preferable to if len(seq): and if not len(seq): respectively.
https://google.github.io/styleguide/pyguide.html#2144-decision